apps/lib/app_provider.c

```diff
@@ -41,6 +46,24 @@ static int opt_provider_load(const char *provider)
     return 1;
 }
 
+OPENSSL_CTX *opt_provider_libctx(void)
+{
+    /*
+     * Load the NULL provider into the default library context and create a
+     * library context which will then be used for any OPT_PROV options.
+     */
+    if (libctx == NULL) {
```

Comment:
Should there be an error if `libctx != NULL`?
I tend to think so.

Reply:
Moved it into app.. This is allowed to be NULL.